RavenDB-13650 - Fixing needless usage of ToDictionary() for simple ca…

ayende opened the pull request:
…lls.

RavenDB-14196 - Fixing missing int cast on method call arguments

ayende committed:
RavenDB-13650 - Fixing needless usage of ToDictionary() for simple calls.
RavenDB-14196 - Fixing missing int cast on method call arguments

arekpalinski reviewed (commented):
It's hard to spot an issue with that change so it looks good to me. Although some tests has failed.

ayende committed:
Merge branch 'RavenDB-14196' of https://github.com/ayende/ravendb into v4.2

ayende committed:
RavenDB-14196
- Avoid emitting cast if there is a different method overload that we want to pick up on the server
- Avoid NRE
